### DISCUSSION TITLE
Orange-Multitarget

### DISCUSSION THREAD

**Discussion by aryt13** [biolab/orange3#5452] [Q&A]
Anyone else getting this error while installing this add-on with Orange 3?

ERROR: Could not find a version that satisfies the requirement Orange-Multitarget
ERROR: No matching distribution found for Orange-Multitarget

https://pypi.org/project/Orange-Multitarget/

**Comment by janezd** [comment]
This is so ancient  that I was initially sure that it's some 3rd party add-on. But now I see it was written by the core group and last touched nine years ago. It works with Orange 2, which is long gone. I can't even find the primary author of this extension -- it was on bitbucket but we apparently removed it. (mpolaj? mlevar?)

Question to others: do we remove this from pipy, too, or just write a comment that this is left here only for the amusement of future archeologists?

**Reply by aryt13** [reply]
That's a pity. :( 
Multi-target variables would have been very useful. 
May I know which version of Orange 2 would this work with? 

I tried to apply it to the following version with no luck...
"orange-win-w-python-1.0-py2.5"

Alternatively, is there any other function within Orange 3 that I could use?

I have a complete data for a certain sample in form of 7 xy curves (single x,multi-y). I want to train a neural network with this complete data and use the outcome to predict missing data points for another sample for which I only have 2 of the xy curves.

**Reply by janezd** [reply]
I've no idea which version of Orange to use... Sorry. Until this morning I forgot this thing existed at all. It is also possible that it's quite buggy -- it's marked as beta.

With Orange 3, you can try training for one output at a time -- just use Select Columns to pick a single target. With neural networks, though, it may be better to train for all at once, but Orange can't do it.

**Reply by aryt13** [reply]
![da](https://user-images.githubusercontent.com/84706515/119342903-49a58f00-bcc8-11eb-863a-5baf7d2abd43.jpg)

Thanks. Maybe I am misunderstanding this, but I don't see how training one output at a time would work. The data looks like this. For a certain material, I have all the data for all the angles. This would be used for training. For other materials, I only have the 0 and 90 degree curves, and the curves in between should be predicted. So in other words, the 0 is the upper bound and 90 is the lower bound. Without having knowledge of both at the same time, I don't think it's possible to predict the values in between. Am I wrong?

**Reply by janezd** [reply]
I guess I misunderstood your problem.

But as I understand now, you are trying to predict the behaviour of y_i (where i is 15, 30, 45, 60 or 75) from the known strain, y_0 and y_90?

**Reply by aryt13** [reply]
Correct. 
and I also have complete data set for all the angles for another material which could be used for the initial training. Because the relationship as you see is not linear (e.g. 45 degree curve is not half way between 0 and 90)

**Reply by janezd** [reply]
But in this case you can use all the known data as "attributes" (features, independent variables) and the one that you try to predict as the target, one at the time.

However, you are predicting curves. Orange doesn't have anything suitable for this. You also have a certain expected shape. I suppose you should be looking either for something like Gaussian processes or some simpler kind of curve fitting.

But again, I may have gotten it wrong again. :)
